src/c++/perf_analyzer/genai-pa/genai_pa/llm_metrics.py

```diff
@@ -262,9 +299,58 @@ def _collect_llm_metrics(
             time_to_first_tokens,
             inter_token_latencies,
             output_token_throughputs,
+            num_generated_tokens,
         )
 
-    def get_statistics(self, infer_mode: str, load_level: int | float) -> Statistics:
-        if (infer_mode, load_level) not in self._profile_results:
-            raise KeyError(f"Profile with {infer_mode}={load_level} does not exist.")
-        return self._profile_results[(infer_mode, load_level)]
+    def _preprocess_response(
+        self, res_timestamps: list[int], res_outputs: list[dict[str, str]]
+    ) -> None:
+        """Helper function to preprocess responses of a request."""
+        if self._service_kind == "openai":
+            openai_final_response = res_outputs[-1]["response"]
+            openai_final_response = remove_sse_prefix(openai_final_response)
+
+            # remove the null final response because this has no output token
+            # and will cause problems when calculating metrics/statistics.
+            if openai_final_response == "[DONE]":
+                res_timestamps.pop()
+                res_outputs.pop()
```

Comment:
@tgerdesnv This is the main thing that got changed. This was needed because the last response from OpenAI is a null final response and causes problem when calculating inter token latency. As a result, I just preprocess and remove it beforehand.